Examples/ObjectiveC/LiveDataExample.m

```diff
@@ -0,0 +1,53 @@
+
+#import "LiveDataExample.h"
+@import Mapbox;
+
+NSString *const MBXExampleLiveData = @"LiveDataExample";
+
+@interface LiveDataExample () <MGLMapViewDelegate>
+
+@property (nonatomic, strong, nullable) NSTimer *timer;
+@property MGLShapeSource *source;
+@end
+
+@implementation LiveDataExample
+
+- (void)viewDidLoad {
+    [super viewDidLoad];
+    
+    // Create a new map view using the Mapbox Dark style.
+    MGLMapView *mapView = [[MGLMapView alloc] initWithFrame:self.view.bounds styleURL:[MGLStyle darkStyleURL]];
+    mapView.autoresizingMask = UIViewAutoresizingFlexibleHeight | UIViewAutoresizingFlexibleWidth;
+    mapView.tintColor = [UIColor darkGrayColor];
+    
+    // Set the map view‘s delegate property.
+    mapView.delegate = self;
+    
+    [self.view addSubview:mapView];
+}
+
+- (void)mapView:(MGLMapView *)mapView didFinishLoadingStyle:(MGLStyle *)style {
+    // Add a source to the map. https://wanderdrone.appspot.com/ generates coordinates for simulated paths.
+    NSURL *url = [NSURL URLWithString:@"https://wanderdrone.appspot.com/"];
+    _source = [[MGLShapeSource alloc] initWithIdentifier:@"wanderdrone-source" URL:url options:nil];
+    [style addSource:_source];
+    
+    // Add a Maki icon to the map to represent the drone's coordinate. For more information about Maki icons, see https://www.mapbox.com/maki-icons/
+    MGLSymbolStyleLayer *droneLayer = [[MGLSymbolStyleLayer alloc] initWithIdentifier:@"wanderdrone-layer" source:_source];
+    droneLayer.iconImageName = [MGLStyleValue valueWithRawValue:@"rocket-15"];
+    [style addLayer:droneLayer];
+    
+    // Create a timer that calls the `updateUrl` function every 1.5 seconds.
+    _timer = [NSTimer scheduledTimerWithTimeInterval:1.5 target:self selector:@selector(updateURL) userInfo:nil repeats:YES];
```

Comment:
If the developer extends this example to switch styles, it’s possible that `_timer` may already be non-nil by this point. Setting it to a different value will leak the previous timer and this view controller. Similar to `-viewWillDisappear:`, this method also needs to call `-invalidate` on `_timer` before setting it to the new value.